framework/src/transfers/MultiAppFieldTransfer.C

```diff
@@ -131,6 +132,9 @@ MultiAppFieldTransfer::transfer(FEProblemBase & to_problem, FEProblemBase & from
     if ((to_mesh.n_nodes() != from_mesh.n_nodes()) || (to_mesh.n_elem() != from_mesh.n_elem()))
       mooseError("The meshes must be identical to utilize MultiAppCopyTransfer.");
 
+    if (!to_problem.coordTransform().isIdentity() || !from_problem.coordTransform().isIdentity())
+      mooseError("Non-identity transformations cannot be used with MultiAppCopyTransfer");
```

Comment:
I'd make it a warning. 
If the meshes are the same there's a high chance the user knows what they are doing and that the node id mappings are correct

Reply:
I would say that instead we should keep it as an error, but we should only error if there is rotation. Post-rotation I don't see how there's any way a user can be confident that the rotated node numberings match

Reply:
well if it's just XY to YZ for example a smart user could know what they're doing and have the same mesh in both apps still

Reply:
I will eat my hat if a user has rotation in one or more of their apps and they are able to get matching dof object numbering post-rotation

Reply:
I am a libMesh developer and I would not be able to do it without a lot of thought. That would be one damn smart user

Reply:
umm I m thinking just load the exact same mesh using a FileMeshGenerator, then rotate one using the coord transform

Reply:
the exact same mesh? Then post-rotation, they do not 100% overlap which is the predicate of the copy transfer

Reply:
but they have matching IDs so it could be done. Error is fine too, was just advocating for non-developing power users, a small group

Reply:
I will make it a warning